File: Assets/UnityChan2D/Demo/FSharpScripts/TimeController.fs
namespace UnityChan2D.Demo
open System.Collections
open UnityEngine

type TimeController () =
  inherit MonoBehaviour ()

  [<DefaultValue>] val mutable public time : int
  [<SceneName>]
  [<DefaultValue>] val mutable public nextLevel : string
  [<DefaultValue>] val mutable public timer : GUIText

  member this.Update () = 
    let remainingTime = this.time - Mathf.FloorToInt(Time.timeSinceLevelLoad * 2.5f)

    if (0 <= remainingTime) then
      this.timer.text <- remainingTime.ToString("000")
    else
      let player = GameObject.FindGameObjectWithTag("Player")
      if player <> null then
        this.StartCoroutine(this.LoadNextLevel()) |> ignore
        this.enabled <- false

  member this.LoadNextLevel () : IEnumerator = 
    let player = GameObject.FindGameObjectWithTag("Player")

    if player <> null then
      player.SendMessage("TimeOver", SendMessageOptions.DontRequireReceiver)

    seq {
      yield WaitForSeconds(3.f)
      Application.LoadLevel(this.nextLevel)
    } :?> IEnumerator

File: Assets/UnityChan2D/Demo/Scripts/TimeController.cs
using UnityEngine;
using System.Collections;
using UniRx;

public class TimeController : ObservableMonoBehaviour
{
    public int time;
    [SceneName]
    public string nextLevel;

    public GUIText timer;

    public override void Awake()
    {
        this.UpdateAsObservable()
            .Subscribe(_ => {
                int remainingTime = time - Mathf.FloorToInt(Time.timeSinceLevelLoad * 2.5f);

                if (0 <= remainingTime)
                {
                    timer.text = remainingTime.ToString("000");
                }
                else
                {
                    GameObject player = GameObject.FindGameObjectWithTag("Player");
                    if (player)
                    {
                        StartCoroutine(LoadNextLevel());
                        enabled = false;
                    }
                }
            });

        base.Awake();
    }

    private IEnumerator LoadNextLevel()
    {
        var player = GameObject.FindGameObjectWithTag("Player");

        if (player)
        {
            player.SendMessage("TimeOver", SendMessageOptions.DontRequireReceiver);
        }

        yield return new WaitForSeconds(3);

        Application.LoadLevel(nextLevel);
    }
}
